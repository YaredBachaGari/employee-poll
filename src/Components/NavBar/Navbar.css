.nav-container {
  display: grid;
  grid-template-columns: 70% 30%;
  align-items: center;
  padding: 1rem;
  border-bottom: 0.5px solid #737272;
}
.nav-menu {
  display: flex;
}
.nav-menu .link {
  margin: 0 1.5rem;
  color: #737272;
}
.nav-menu .link:hover {
    color: brown;
    text-decoration: underline 3px solid;
  }
.avatar-logout .logout {
  background: none;
  font-size: 1.2rem;
  font-weight: bold;
  color: #737272;
}
.avatar-logout .logout:hover {
  cursor: pointer;
  color: brown;
}
.nav-menu {
  font-size: 1.2rem;
  font-weight: bold;
}
.avatar-logout {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}
.avatar-username{
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: flex-end;
}
.profileAvatar{
    display: flex;
    justify-content: center
}