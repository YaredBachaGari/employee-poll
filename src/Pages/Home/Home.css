.poll-category {
  width: 75%;
  margin: 5rem auto;
}
.loadingmsg {
  margin: 5rem auto;
  text-align: center;
  font-size: 2rem;
  font-weight: bold;
}
.toggler {
  margin: 3rem auto;
  text-align: center;
}
.defaultToggle {
  text-decoration: underline solid 2px brown;
}
.toggler button {
  background: none;
  border: none;
  padding: 0.5rem 1rem;
  margin: 0 2rem;
  font-size: 1.25rem;
  font-weight: bold;
  color: #737373;
}
.toggler button:hover {
  cursor: pointer;
  color: brown;
}
