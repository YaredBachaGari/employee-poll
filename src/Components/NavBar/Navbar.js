import React from "react";
import { Link } from "react-router-dom";
import Avatar from "../Avatar/Avatar";
import "./Navbar.css";
import { useNavigate } from "react-router-dom";
import { useDispatch } from "react-redux";
import { logoutUser } from "../../Redux-handler/Actions/AuthUser";

const Navbar = ({ AuthUser }) => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const username = AuthUser?.loggedInUser?.username;
  const avatarURL = AuthUser?.loggedInUser?.avatarURL;

  const handleLogout = () => {
    dispatch(logoutUser());
    navigate("/");
  };
  return (
    <div className="nav-container">
      <div className="nav-menu">
        <Link to="/home" className="link" data-testid="home">
          Home
        </Link>
        <Link to="/leaderboard" className="link" data-testid="leaderBoard">
          LeaderBoard
        </Link>
        <Link to="/add" className="link" data-testid="newpoll">
          New
        </Link>
      </div>
      <div className="avatar-logout">
        <div className="avatar-username profileAvatar">
          <Avatar AvatarUrl={avatarURL} />
          <span data-testid="username">{username}</span>
        </div>
        <button
          data-testid="logout"
          onClick={() => {
            handleLogout();
          }}
        >
          Logout
        </button>
      </div>
    </div>
  );
};

export default Navbar;
