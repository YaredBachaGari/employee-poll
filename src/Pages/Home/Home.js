import React from "react";
import Navbar from "../../Components/NavBar/Navbar";
import PollCointainer from "../../Components/PollCointainer/PollCointainer";
import "./Home.css";
import HOC from "../../Components/HigherOrderComp/HOC.js";
import { filterData } from "../../utils/helper";

const Home = ({ allUsers, AuthUser, Questions }) => {
  const filteredQ = filterData(Questions, AuthUser);
  const newQuestions = filteredQ.new;
  const completed = filteredQ.complete;
  return (
    <>
      {Questions.loading || allUsers.loading ? (
        <p className="loadingmsg">Loading...</p>
      ) : (
        <div>
          <Navbar AuthUser={AuthUser} />
          <div className="poll-category">
            <PollCointainer
              category={"New Questions"}
              allUsers={allUsers}
              AuthUser={AuthUser}
              data={newQuestions}
            />
          </div>
          <div className="poll-category">
            <PollCointainer
              category={"Done"}
              allUsers={allUsers}
              AuthUser={AuthUser}
              data={completed}
            />
          </div>
        </div>
      )}
    </>
  );
};

export default HOC(Home);
