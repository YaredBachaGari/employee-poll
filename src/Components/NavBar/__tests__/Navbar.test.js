import React from "react";
import { BrowserRouter, MemoryRouter, Routes, Route } from "react-router-dom";
import configureStore from "redux-mock-store";
//import { logout } from "../../store/actions/AuthUser";
import store from "../../../Redux-handler/Store/Store";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import Navbar from "../Navbar";
import { render, screen, fireEvent } from "@testing-library/react";
import Home from "../../../Pages/Home/Home";

describe("Navbar component", () => {
  it("it should match the snapshots without any prop", () => {
    const { navigation } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>
    );
    expect(navigation).toMatchSnapshot();
  });
  it("it should match the snapshots with prop being passed", () => {
    const mockStore = configureStore();
    const store = mockStore({
      AuthUser: {
        loggedInUser: { id: 1, username: "john" },
      },
    });
    const { navigation } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>
    );
    expect(navigation).toMatchSnapshot();
  });
  it("should render all menu links", () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar AuthUser={store.AuthUser} />
        </BrowserRouter>
      </Provider>
    );
    expect(screen.getByTestId("home")).toBeInTheDocument();
    expect(screen.getByTestId("leaderBoard")).toBeInTheDocument();
    expect(screen.getByTestId("newpoll")).toBeInTheDocument();
    expect(screen.getByTestId("logout")).toBeInTheDocument();
  });
});
describe("Navbar component2", () => {
  it("Avatar and user name should render on navbar", () => {
    const mockStore = configureStore();
    const store = mockStore({
      AuthUser: {
        loggedInUser: { id: 1, username: "john" },
      },
    });

    render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar AuthUser={store.AuthUser} />
        </BrowserRouter>
      </Provider>
    );
    expect(screen.getByTestId("username")).toBeInTheDocument();
    expect(screen.getByAltText("avatar")).toBeInTheDocument();
  });
});
