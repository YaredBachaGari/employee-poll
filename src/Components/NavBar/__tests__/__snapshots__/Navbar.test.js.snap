// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component it should match the snapshots with prop being passed 1`] = `undefined`;

exports[`Navbar component it should match the snapshots without any prop 1`] = `undefined`;
